Model/SourcesRepository.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Model;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\SourcesFactory;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Model\Sources as SourceModel;
+use Magento\Framework\Serialize\SerializerInterface;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionItemInterfaceFactory;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var \Magento\Framework\Api\DataObjectHelper
+     */
+    protected $dataObjectHelper;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourceSelectionItemInterfaceFactory
+     */
+    private $sourceSelectionItemInterface;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SerializerInterface $serializer
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SerializerInterface $serializer,
+        \Magento\Framework\Api\DataObjectHelper $dataObjectHelper,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->serializer = $serializer;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourceSelectionItemInterface = $sourceSelectionItemInterface;
+
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
```

Comment:
Yo, can we throw a not found error here if the source extension entry does not exist?